public/js/index.js

```diff
@@ -296,6 +296,64 @@ async function addUser(user) {
 }
 
 async function addLesson() {
+	const content = quill.root.innerHTML;
+	const isEditView = createLessonContainer
+		.getAttribute('view')
+		.includes('edit-lesson');
+	// Regex to match any number of whitespaces in the content form.
+	var regex = /<(.|\n)*?>/g;
+	if (content.replace(regex, '').trim().length === 0) {
+		console.log('Tried to add empty lesson note.');
+		return;
+	}
+	if (isEditView) {
+		try {
+			const id = createLessonContainer.getAttribute('view').split(':')[1];
+
+			const snapshot = await db
+				.collection('users')
+				.doc(firebase.auth().currentUser.uid)
+				.get();
+
+			const newLessons = snapshot.data().lessons.map((lesson) => {
+				if (lesson.id === id) {
+					lesson.title = lessonInput.value;
+					lesson.content = content;
+				}
+				return lesson;
+			});
+
+			if (!newLessons.length) {
+				console.log('Tried to add empty lessons.');
+				return;
+			}
+
+			await snapshot.ref.update({ lessons: newLessons });
+			renderLessons();
+			createLessonContainer.setAttribute('view', 'create-lesson');
+		} catch (error) {
+			console.log(error);
+		}
+	} else {
+		try {
+			await db
+				.collection('users')
+				.doc(firebase.auth().currentUser.uid)
+				.update({
+					lessons: firebase.firestore.FieldValue.arrayUnion({
+						title: lessonInput.value,
+						content,
+						id: String(Math.floor(Math.random() * 90000 + 10000)),
+					}),
+				});
+
+			lessonInput.value = '';
+			renderLessons();
+		} catch (error) {
+			console.log(error);
+		}
+	}
+
   const content = quill.root.innerHTML;
```

Comment:
Looks like we have some duplicate code within `addLesson()` that needs removing

Reply:
Done! Removed the duplicate code within addLesson() and commited the changes. 
Let me know if there's anything more.

Reply:
Perfect!